addons/talo/apis/leaderboards_api.gd

```diff
@@ -1,77 +1,97 @@
-class_name LeaderboardsAPI extends TaloAPI
+## LeaderboardsAPI
+extends TaloAPI
 ## An interface for communicating with the Talo Leaderboards API.
 ##
 ## This API is used to read and update leaderboards in your game. Leaderboards are used to track player scores and rankings.
 ##
 ## @tutorial: https://docs.trytalo.com/docs/godot/leaderboards
 
-var _entries_manager = TaloLeaderboardEntriesManager.new()
+var _entries_manager := TaloLeaderboardEntriesManager.new()
 
 ## Get a list of all the entries that have been previously fetched or created for a leaderboard.
 func get_cached_entries(internal_name: String) -> Array:
 	return _entries_manager.get_entries(internal_name)
 
 ## Get a list of all the entries that have been previously fetched or created for a leaderboard for the current player.
-func get_cached_entries_for_current_player(internal_name: String) -> Array:
+func get_cached_entries_for_current_player(internal_name: String) -> Array[TaloLeaderboardEntry]:
 	if Talo.identity_check() != OK:
 		return []
 
 	return _entries_manager.get_entries(internal_name).filter(
-		func (entry: TaloLeaderboardEntry):
+		func(entry: TaloLeaderboardEntry) -> bool:
 			return entry.player_alias.id == Talo.current_alias.id
 	)
 
 ## Get a list of entries for a leaderboard. The page parameter is used for pagination.
-func get_entries(internal_name: String, page: int, alias_id = -1, include_archived = false) -> Array:
-	var url = "/%s/entries?page=%s"
-	var url_data = [internal_name, page]
+func get_entries(internal_name: String, page: int, alias_id := -1, include_archived := false) -> EntriesPage:
```

Comment:
great change adding these structs - would be amazing if you could also pull this out into its own PR